internal/experiment/generator.go

```diff
@@ -103,7 +109,30 @@ func (g *Generator) SetDefaultSelectors() {
 
 // Execute the experiment generation pipeline, sending the results to the supplied writer.
 func (g *Generator) Execute(output kio.Writer) error {
+	scenario, err := application.GetScenario(&g.Application, g.Scenario)
+	if err != nil {
+		return err
+	}
+
+	objective, err := application.GetObjective(&g.Application, g.Objective)
+	if err != nil {
+		return err
+	}
+
+	// Compute the effective scenario, objective, and experiment names
+	scenarioName, objectiveName, experimentName := "", "", g.ExperimentName
+	if scenario != nil {
+		scenarioName = scenario.Name
+	}
+	if objective != nil {
+		objectiveName = objective.Name
+	}
```

Comment:
It seems like we're pushing this down through a few layers instead of just failing early in GetScenario/GetObjective if these are nil. Im expecting if scenario/objective are nil we're dealing with an invalid application, so would this cause issues further down in the kio pipeline?

Reply:
I added validation to the experiment (in the transfomer) to ensure we do not produce invalid experiments (i.e. you need a parameter and a metric) since that is what is ultimately matters (e.g. `redskyctl generate experiment` fails because there are no parameters on the generated experiment). By that argument, not having a scenario is more viable then not having an objective (since an experiment is still valid without a trial job). I would rather not force the validation here to leave the door open for things like the ability to pull metric definitions while scanning.